tests/topotests/bgp_distance_change/test_bgp_admin_dist.py

```diff
@@ -0,0 +1,1279 @@
+
+
+def teardown_module(mod):
+    """teardown_module.
+
+    Teardown the pytest environment.
+    * `mod`: module name
+    """
+    logger.info("Running teardown_module to delete topology")
+    tgen = get_topogen()
+
+    # Stop toplogy and Remove tmp files
+    tgen.stop_topology()
+
+    logger.info(
+        "Testsuite end time: {}".format(time.asctime(time.localtime(time.time())))
+    )
+    logger.info("=" * 40)
+
+
+#####################################################
+# Tests starting
+#####################################################
+def test_bgp_admin_distance_ebgp_ecmp_p0():
+    """
+    TC: 2
+    Verify ebgp admin distance functionality with ECMP.
+    """
+    tgen = get_topogen()
+    global bgp_convergence
+
+    if bgp_convergence is not True:
+        pytest.skip("skipped because of BGP Convergence failure")
+
+    # test case name
+    tc_name = inspect.stack()[0][3]
+    write_test_header(tc_name)
+    if tgen.routers_have_failure():
+        check_router_status(tgen)
+
+    step("Configure base config as per the topology")
+    reset_config_on_routers(tgen)
+
+    step("Configure static route 50.0.20.1/32 in R4 and R5, redistribute in bgp")
+
+    for addr_type in ADDR_TYPES:
+
+        input_dict = {
+            "r4": {
+                "static_routes": [{"network": NETWORK[addr_type], "next_hop": "Null0"}]
+            }
+        }
+
+        result = create_static_routes(tgen, input_dict)
+        assert result is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result
+        )
+
+    for addr_type in ADDR_TYPES:
+
+        input_dict = {
+            "r5": {
+                "static_routes": [{"network": NETWORK[addr_type], "next_hop": "Null0"}]
+            }
+        }
+
+        result = create_static_routes(tgen, input_dict)
+        assert result is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result
+        )
+
+    step("Verify that route is learnt in DUT via ebgp")
+
+    # Verifying RIB routes
+    protocol = "bgp"
+    input_dict = topo["routers"]
+    dut = "r3"
+    nhop = {"ipv4": [], "ipv6": []}
+    nhop["ipv4"].append(topo["routers"]["r4"]["links"]["r3"]["ipv4"].split("/")[0])
+    nhop["ipv4"].append(topo["routers"]["r5"]["links"]["r3"]["ipv4"].split("/")[0])
+    nhop["ipv6"].append(get_frr_ipv6_linklocal(tgen, "r4", "r3-r4-eth1"))
+    nhop["ipv6"].append(get_frr_ipv6_linklocal(tgen, "r5", "r1-r3-eth1"))
+
+    for addr_type in ADDR_TYPES:
+        input_dict = {
+            "r3": {
+                "static_routes": [
+                    {"network": NETWORK[addr_type][0], "next_hop": "Null0"}
+                ]
+            }
+        }
+        result4 = verify_rib(tgen, addr_type, dut, input_dict, protocol=protocol)
+        assert result4 is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result4
+        )
+
+    step("Configure the static route 50.0.20.1/32 in R3 (Dut).")
```

Comment:
Wrong IP?

Reply:
Fixed in all the places.

Reply:
Fixed. Kindly check.  Thank you @ton31337